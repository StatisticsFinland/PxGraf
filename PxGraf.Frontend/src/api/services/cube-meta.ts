/* istanbul ignore file */

import ApiClient from "api/client";
import { useQuery } from "react-query";
import { ICubeMeta } from "types/cubeMeta";
import { defaultQueryOptions } from "utils/ApiHelpers";

/**
 * Interface for cube metadata result. Cube is a multi-dimensional result of Px file data filtered by a query.
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred during loading.
 * @property {ICubeMeta} data - The cube meta data as an @see {@link ICubeMeta} object.
 */
export interface ICubeMetaResult {
    isLoading: boolean;
    isError: boolean;
    data: ICubeMeta;
}

const fetchCubeMeta = async(idStack: string[]): Promise<ICubeMeta> => {
    const client = new ApiClient();
    const url = `creation/cube-meta/${idStack.join('/')}`;

    return await client.getAsync(url);
}

export const useCubeMetaQuery = (idStack: string[]): ICubeMetaResult => {
    return useQuery(
        ['cube-meta', idStack],
        () => fetchCubeMeta(idStack),
        defaultQueryOptions
    );
}