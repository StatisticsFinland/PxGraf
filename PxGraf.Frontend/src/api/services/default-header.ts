/* istanbul ignore file */

import ApiClient from "api/client";
import { merge } from "lodash";
import { useQuery } from "react-query";
import { MultiLanguageString } from "types/multiLanguageString";
import { Query } from "types/query";
import { buildTableReference, defaultQueryOptions } from "utils/ApiHelpers";

/**
 * Interface for the chart's header result
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred while retrieving the header.
 * @property {MultiLanguageString} data - The header data represented as a multi-language string.
 */
export interface IHeaderResult {
    isLoading: boolean;
    isError: boolean;
    data: MultiLanguageString;
}

const fetchDefaultHeader = async(idStack: string[], query: Query): Promise<MultiLanguageString> => {
    const client = new ApiClient();
    const requestBody = JSON.stringify(
        merge(
            { tableReference: buildTableReference(idStack) },
            { variableQueries: query }
        )
    );
  
    const url = 'creation/default-header';
    return await client.postAsync(url, requestBody);
}

export const useDefaultHeaderQuery = (idStack: string[], query: Query): IHeaderResult => {
    return useQuery(
        ['default-header', ...idStack, query],
        () => fetchDefaultHeader(idStack, query),
        {
            ...defaultQueryOptions,
            enabled: query != null,
        }
    );
}