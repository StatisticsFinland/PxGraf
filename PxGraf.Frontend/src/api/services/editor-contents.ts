/* istanbul ignore file */

import ApiClient from "api/client";
import { useQuery } from "react-query";
import { IEditorContentsResponse } from "types/editorContentsResponse";
import { buildCubeQuery, defaultQueryOptions, useDebounceState } from "utils/ApiHelpers";
import { ICubeQuery, Query } from "types/query";

/**
 * Interface for editor contents result. Editor contents are used by the editor to populate the settings for adjusting the visualization.
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred during loading.
 * @property {IEditorContentsResponse} data - The editor contents as an @see {@link IEditorContentsResponse} object.
 */ 
export interface IEditorContentsResult {
    isLoading: boolean;
    isError: boolean;
    data: IEditorContentsResponse;
}

const fetchEditorContents = async (idStack: string[], query: Query, metaEdits: ICubeQuery): Promise<IEditorContentsResponse> => {
    const client = new ApiClient();
    const url = 'creation/editor-contents';
    const requestBody = JSON.stringify(buildCubeQuery(query, metaEdits, idStack));
    return await client.postAsync(url, requestBody);
}

export const useEditorContentsQuery = (idStack: string[], query: Query, cubeQuery: ICubeQuery): IEditorContentsResult => {
    [idStack, query, cubeQuery] = useDebounceState(1000, idStack, query, cubeQuery);
    const queryKey = ['editor-contents', ...idStack, query, cubeQuery];
    return useQuery(
        queryKey,
        () => fetchEditorContents(idStack, query, cubeQuery),
        {
            ...defaultQueryOptions,
            enabled: query != null,
            keepPreviousData: true,
        }
    );
}