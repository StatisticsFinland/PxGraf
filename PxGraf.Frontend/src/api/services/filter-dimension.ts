/* istanbul ignore file */

import ApiClient from "api/client";
import { useQuery } from "react-query";
import { IValueFilter, Query } from "types/query";
import { buildTableReference, defaultQueryOptions } from "utils/ApiHelpers";

/**
 * Interface for variable filter result
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred during loading.
 * @property {{ [key: string]: string[] }} data - The variable filter data represented by a dictionary. Key is the variable code and value is a list of strings that represent value codes for that variable.
 */
export interface IFilterVariableResult {
    isLoading: boolean;
    isError: boolean;
    data: {[key: string]: string[]}
}

const fetchResolveVariableFilter = async (idStack: string[], variableFilters: { [key: string]: IValueFilter }): Promise<{ [key: string]: string[] }> => {
    const client = new ApiClient();
    const requestBody = JSON.stringify(
        {
            tableReference: buildTableReference(idStack),
            filters: variableFilters,
        }
    );

    const url = 'creation/filter-dimension'
    return await client.postAsync(url, requestBody);
}

export const useResolveVariableFiltersQuery = (idStack: string[], query: Query): IFilterVariableResult => {
    const varQueries = Object.entries(query ?? {});
    const varFilters =  Object.fromEntries(varQueries.map(([variableCode, variableQuery]) => {
        return [variableCode, variableQuery.valueFilter]
    }));

    return useQuery(
        ['filter-dimension', ...idStack, query],
        () => fetchResolveVariableFilter(idStack, varFilters),
        defaultQueryOptions
    );
}