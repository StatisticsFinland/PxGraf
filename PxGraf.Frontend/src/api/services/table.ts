/* istanbul ignore file */

import ApiClient from "api/client";
import { useQuery } from "react-query";
import { defaultQueryOptions } from "utils/ApiHelpers";
import { MultiLanguageString } from "../../types/multiLanguageString";

/**
 * Interface for a table list response.
 * @property {string} id - The id of the table.
 * @property {string} type - The type of the table where 't' is a table and 'l' is a list item.
 * @property {string} updated - The last updated date of the table.
 * @property {MultiLanguageString} text - The description of the table as a multi-language string.
 * @property {string[]} languages - Available languages of the table.
 */
export interface ITableListResponse {
    id: string;
    type?: 't' | 'l';
    updated?: string;
    text: MultiLanguageString;
    languages?: string[];
}

/**
 * Interface for a table result.
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred during loading.
 * @property {ITableListResponse[]} data - The table list response data.
 */
export interface ITableResult {
    isLoading: boolean;
    isError: boolean;
    data: ITableListResponse[];
}

const fetchTable = async (idStack: string[], lang: string): Promise<ITableListResponse[]> => {
    const client = new ApiClient();
    const path = idStack.join("/");
    const url = "creation/data-bases/" + path;
    const getParams = {
        'lang': lang
    }

    return await client.getAsync(url, getParams);
};

export const sortTableData = (data: ITableListResponse[], primaryLanguage: string, databaseLanguages: string[]): ITableListResponse[] => {
    return [...data].sort((a, b) => {
        const textA = a.text[primaryLanguage] || a.text[databaseLanguages[0]];
        const textB = b.text[primaryLanguage] || b.text[databaseLanguages[0]];

        return textA.localeCompare(textB, primaryLanguage);
    });
};

export const useTableQuery = (idStack: string[], lang: string): ITableResult => {
    // If trying to fetch a database without a language, return empty
    const fetchFunction = lang ?
        () => fetchTable(idStack, lang) :
        () => Promise.resolve([]);

    return useQuery(
        ['table', ...idStack, lang],
        fetchFunction,
        defaultQueryOptions,
    );
};