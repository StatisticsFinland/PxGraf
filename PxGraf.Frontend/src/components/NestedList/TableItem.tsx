import { Alert, AlertTitle, Divider, ListItem, ListItemButton, ListItemIcon, ListItemText } from "@mui/material";
import FileIcon from '@mui/icons-material/InsertDriveFileOutlined';
import React from 'react';
import { Link } from "react-router-dom";
import { useTranslation } from "react-i18next";
import styled from "styled-components";
import { urls } from "Router";
import { UiLanguageContext } from "contexts/uiLanguageContext";
import { parseLanguageString } from 'utils/ApiHelpers';
import { IDatabaseTable } from 'types/tableListItems';

const StyledListItem = styled(ListItem)`
  background-color: #f8f8f8;
`;

const ErrorAlert = styled(Alert)`
  width: 100%;
`;

/**
 * @property {string[]} currentPath Path to the table in in the hierarchy of the table list.
 * @property {IDatabaseTable} item Contains information about the table.
 * @property {number} depth Current depth in the hierarchy of the table list.
 */
interface ITableItemProps {
    currentPath: string[];
    item: IDatabaseTable;
    depth: number;
}

/**
 * Component used for displaying a table in @see {@link NestedList} component.
 */
export const TableItem: React.FC<ITableItemProps> = ({ currentPath, item, depth }) => {
    const { t } = useTranslation();
    const { language } = React.useContext(UiLanguageContext);
    const displayLanguage = item.languages.includes(language) ? language : item.languages[0];

    return <React.Fragment key={`${item.fileName}-key`}>
        <StyledListItem id={currentPath.join('-')}>
            <ListItemButton sx={{ mr: 3, pl: depth * 4 }} component={Link} to={urls.editor(currentPath)}>
                {item.error ?
                    <ErrorAlert sx={{ pl: depth * 4 }} severity="warning">
                        <AlertTitle>{`${item.name[displayLanguage] ?? item.fileName}`}</AlertTitle>
                        {t("error.contentVariableMissing")}
                    </ErrorAlert>
                    :
                    <React.Fragment>
                        <ListItemIcon sx={{ minWidth: '32px' }}>
                            <FileIcon />
                        </ListItemIcon>
                        <ListItemText primary={`${item.name[displayLanguage]} ${parseLanguageString(item.languages)}`} secondary={t("tableSelect.updated") + ": " + new Date(item.lastUpdated).toLocaleString(language)} />
                    </React.Fragment>
                }
            </ListItemButton>
        </StyledListItem>
        <Divider />
    </React.Fragment>
}