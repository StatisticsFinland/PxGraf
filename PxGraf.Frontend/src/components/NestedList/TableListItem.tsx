import * as React from 'react';
import { Link } from "react-router-dom";
import { useTranslation } from 'react-i18next';

import { List, ListItem, ListItemIcon, ListItemText, ListItemButton, Divider, IconButton, Collapse, Tooltip } from '@mui/material';

import ExpandLess from '@mui/icons-material/ExpandLess';
import ExpandMore from '@mui/icons-material/ExpandMore';
import ListViewIcon from '@mui/icons-material/ListAltOutlined';

import { urls } from 'Router';
import NestedList from './NestedList';
import { UiLanguageContext } from 'contexts/uiLanguageContext';
import { parseLanguageString } from 'utils/ApiHelpers';
import { IDatabaseGroupHeader } from 'types/tableListItems';

interface ITableListItemProps {
    currentPath: string[];
    item: IDatabaseGroupHeader;
    initialOpenState?: boolean;
    depth: number;
}

/**
 * Component used for displaying a database or subfolder in @see {@link NestedList} component.
 * @param {string[]} currentPath Path to the item in question in the Px file system.
 * @param {IDatabaseGroupHeader} item Response object that stores information about the database or subfolder.
 * @param {boolean} initialOpenState Whether the item should be displayed as opened.
 * @param {number} depth Current browsing depth.
 */
export const TableListItem: React.FC<ITableListItemProps> = ({ currentPath, item, initialOpenState, depth }) => {
    const { t } = useTranslation();
    const [isOpen, setIsOpen] = React.useState(initialOpenState ?? false);
    const { language } = React.useContext(UiLanguageContext);
    const displayLanguage = item.languages.includes(language) ? language : item.languages[0];

    return <React.Fragment key={`${item.code}-key`}>
        <ListItem
            secondaryAction={depth >= 1 ?
                <Tooltip title={t("tableSelect.listView")} aria-label={`${t("tableSelect.listView")}: ${item.name[displayLanguage]}`}>
                    <IconButton edge="end" component={Link} to={urls.tableList(currentPath)} >
                        <ListViewIcon />
                    </IconButton>
                </Tooltip> : <></>
            }>
            <ListItemButton sx={{ pl: depth * 4 }} onClick={() => setIsOpen(!isOpen)} aria-expanded={isOpen}>
                <ListItemIcon sx={{ minWidth: '32px' }}>
                    {isOpen ? <ExpandLess /> : <ExpandMore />}
                </ListItemIcon>
                {isOpen ? <ListItemText primary={<b>{`${item.name[displayLanguage]} ${parseLanguageString(item.languages)}`}</b>} /> : <ListItemText primary={`${item.name[displayLanguage]} ${parseLanguageString(item.languages)}`} />}
            </ListItemButton>
        </ListItem>
        <Divider />
        <Collapse in={isOpen} timeout="auto" unmountOnExit>
            <List component="div" disablePadding>
                <NestedList path={currentPath} depth={depth + 1} />
            </List>
        </Collapse>
    </React.Fragment>;
};