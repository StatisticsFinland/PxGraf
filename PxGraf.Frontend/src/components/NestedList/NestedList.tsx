import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { ListItem, ListItemIcon, ListItemText, Divider, Skeleton, Alert } from '@mui/material';
import UiLanguageContext from 'contexts/uiLanguageContext';
import styled from 'styled-components';
import { TableItem } from './TableItem';
import { TableListItem } from './TableListItem';
import useHierarchyParams from 'hooks/useHierarchyParams';
import useScrollToElement from 'hooks/useScrollToElement';
import { useTableQuery } from 'api/services/table';
import { IDatabaseGroupHeader, IDatabaseTable } from 'types/tableListItems';
import { sortDatabaseItems } from 'utils/sortingHelpers';

const StyledSkeleton = styled(Skeleton)`
  width: 24px;
  height: 24px;
`;

const StyledListItemText = styled(ListItemText)`
  margin-right: 32px;
`;

const ErrorAlert = styled(Alert)`
  width: 100%;
`;

/**
 * @property {string[]} path - Path to the current location in the tree.
 * @property {number} depth - Current browsing depth.
 */
interface INestedListProps {
    path: string[];
    depth?: number;
}

/**
 * Nested list component for displaying the database and table hierarchies.
 * @see TableListItem component is used for displaying databases and subfolders while @see {@link TableItem} component is used for displaying individual tables.
 */
export const NestedList: React.FC<INestedListProps> = ({ path, depth }) => {
    const { language } = React.useContext(UiLanguageContext);
    const { isLoading, isError, data } = useTableQuery(path);
    const { t } = useTranslation();
    
    const tablePath = useHierarchyParams();
    const activeId: string = tablePath.length && tablePath.join('-');
    const shouldScroll: boolean = activeId && data && path.length === tablePath.length - 1 && tablePath.join(',').startsWith(path.join(','));
    useScrollToElement(shouldScroll && activeId);

    const sortedGroups: IDatabaseGroupHeader[] = React.useMemo(() => {
        if (!data?.headers) return null;
        return sortDatabaseItems(data.headers, language);
    }, [data, language]);

    const sortedTables: IDatabaseTable[] = React.useMemo(() => {
        if (!data?.files) return null;
        return sortDatabaseItems(data.files, language);
    }, [data, language]);

    if (isLoading) {
        return <React.Fragment key={-1}>
            <ListItem>
                <ListItemIcon sx={{ ml: depth * 4 }}>
                    <StyledSkeleton variant="circular" />
                </ListItemIcon>
                <StyledListItemText primary={<Skeleton variant="text" />} />
            </ListItem>
            <Divider />
        </React.Fragment>;
    }
    else if (isError) {
        return <React.Fragment key={-2}>
            <ListItem>
                <ErrorAlert sx={{ pl: depth * 4 }} severity="error">{t("error.contentLoad")}</ErrorAlert>
            </ListItem>
            <Divider />
        </React.Fragment>;
    }

    return (
        <>
            {
                sortedGroups ? sortedGroups.map((item) => (
                    <TableListItem
                        key={`${item.code}-list-key`}
                        currentPath={[...path, item.code]}
                        item={item}
                        initialOpenState={sortedGroups.length < 2 || tablePath.join(',').startsWith([...path, item.code].join(','))}
                        depth={depth ?? 0}
                    />
                )) : <ErrorAlert sx={{ pl: depth * 4 }} severity="error">{t("error.contentLoad")}</ErrorAlert>
            }
            {
                sortedTables ? sortedTables.map((item) => (
                    <TableItem
                        key={`${item.fileName}-table-key`}
                        currentPath={[...path, item.fileName]}
                        item={item}
                        depth={depth ?? 0}
                    />
                )) : <ErrorAlert sx={{ pl: depth * 4 }} severity="error">{t("error.contentLoad")}</ErrorAlert>
            }
        </>
    );
}
export default NestedList;