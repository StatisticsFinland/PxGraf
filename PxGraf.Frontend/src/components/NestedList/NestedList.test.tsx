import React from 'react';
import { render } from '@testing-library/react';
import UiLanguageContext from 'contexts/uiLanguageContext';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import { ITableResult } from '../../api/services/table';
import NestedList from './NestedList';

jest.mock('react-i18next', () => ({
    ...jest.requireActual('react-i18next'),
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => null),
            },
        };
    },
}));

jest.mock('envVars', () => ({
    PxGrafUrl: 'pxGrafUrl.fi/',
    PublicUrl: 'publicUrl.fi/'
}));
const mockTableQueryResult: ITableResult = {
    isLoading: false,
    isError: false,
    data: {
        headers: [
            {
                code: 'asd',
                name: { 'fi': 'foo-fi', 'en': 'foo-en', 'sv': 'foo-sv' },
                languages: ['fi', 'en', 'sv']
            },
            {
                code: 'id',
                name: { 'fi': 'foo3-fi', 'en': 'foo3-en' },
                languages: ['fi', 'en']
            }
        ],
        files: [
            {
                code: 'asd2',
                lastUpdated: '2021-11-13T14:53:06',
                name: { 'fi': 'foo2-fi', 'sv': 'foo2-sv' },
                languages: ['en', 'sv']
            }
        ]
    }
}

jest.mock('api/services/table', () => ({
    ...jest.requireActual('api/services/table'),
    useTableQuery: () => {
        return mockTableQueryResult;
    },
}));

const mockDepth = 3;
const mockPath = ['foo', 'bar', 'baz'];

const setLanguage = jest.fn();
const language = 'fi';

const setLanguageTab = jest.fn();
const languageTab = 'fi';

const availableUiLanguages = ['fi', 'en', 'sv'];

const uiContentLanguage = "fi";
const setUiContentLanguage = jest.fn();

describe('Rendering test', () => {
    it('renders correctly', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <UiLanguageContext.Provider value={{ language, setLanguage, languageTab, setLanguageTab, availableUiLanguages, uiContentLanguage, setUiContentLanguage }}>
                    <NestedList depth={mockDepth} path={mockPath} />
                </UiLanguageContext.Provider>
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('Assertion tests', () => {
    it('shows item listed with the first available if given ui language is not supported', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <UiLanguageContext.Provider value={{ language: 'fr', setLanguage, languageTab, setLanguageTab, availableUiLanguages, uiContentLanguage, setUiContentLanguage }}>
                    <NestedList depth={null} path={mockPath} />
                </UiLanguageContext.Provider>
            </MemoryRouter>
        );
        expect(asFragment().textContent).toContain('foo-fi');
    });
});