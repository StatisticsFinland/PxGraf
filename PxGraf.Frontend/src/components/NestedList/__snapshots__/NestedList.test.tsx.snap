// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering test renders correctly 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p4qrhw-MuiListItem-root"
  >
    <div
      aria-expanded="false"
      class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-nlxtz9-MuiButtonBase-root-MuiListItemButton-root"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root css-sx7q2d-MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
      <div
        class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
        >
          foo-fi (FI, EN, SV)
        </span>
      </div>
    </div>
    <div
      class="MuiListItemSecondaryAction-root css-zezwi5-MuiListItemSecondaryAction-root"
    >
      <a
        aria-label="tableSelect.listView: foo-fi"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
        data-discover="true"
        data-mui-internal-clone-element="true"
        href="/table-list/foo/bar/baz/asd/"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="ListAltOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 7h6v2h-6zm0 4h6v2h-6zm0 4h6v2h-6zM7 7h2v2H7zm0 4h2v2H7zm0 4h2v2H7zM20.1 3H3.9c-.5 0-.9.4-.9.9v16.2c0 .4.4.9.9.9h16.2c.4 0 .9-.5.9-.9V3.9c0-.5-.5-.9-.9-.9M19 19H5V5h14z"
          />
        </svg>
      </a>
    </div>
  </li>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
  />
  <li
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p4qrhw-MuiListItem-root"
  >
    <div
      aria-expanded="false"
      class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-nlxtz9-MuiButtonBase-root-MuiListItemButton-root"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root css-sx7q2d-MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
      <div
        class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
        >
          foo3-fi (FI, EN)
        </span>
      </div>
    </div>
    <div
      class="MuiListItemSecondaryAction-root css-zezwi5-MuiListItemSecondaryAction-root"
    >
      <a
        aria-label="tableSelect.listView: foo3-fi"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
        data-discover="true"
        data-mui-internal-clone-element="true"
        href="/table-list/foo/bar/baz/id/"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="ListAltOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 7h6v2h-6zm0 4h6v2h-6zm0 4h6v2h-6zM7 7h2v2H7zm0 4h2v2H7zm0 4h2v2H7zM20.1 3H3.9c-.5 0-.9.4-.9.9v16.2c0 .4.4.9.9.9h16.2c.4 0 .9-.5.9-.9V3.9c0-.5-.5-.9-.9-.9M19 19H5V5h14z"
          />
        </svg>
      </a>
    </div>
  </li>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
  />
  <li
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding sc-jHatCz bbCcAT css-cfd5wn-MuiListItem-root"
    id="foo-bar-baz-asd2.px"
  >
    <a
      class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1j5cvue-MuiButtonBase-root-MuiListItemButton-root"
      data-discover="true"
      href="/editor/foo/bar/baz/asd2.px/"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root css-sx7q2d-MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="InsertDriveFileOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8zM6 20V4h7v5h5v11z"
          />
        </svg>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
        >
          undefined (EN, SV)
        </span>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
        >
          tableSelect.updated: 13.11.2021 klo 14.53.06
        </p>
      </div>
    </a>
  </li>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
  />
</DocumentFragment>
`;

exports[`Rendering test renders correctly when table query  1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorError MuiAlert-standardError MuiAlert-standard sc-hOTSeL hjjwaJ css-u8pub9-MuiPaper-root-MuiAlert-root"
      role="alert"
      style="--Paper-shadow: none;"
    >
      <div
        class="MuiAlert-icon css-vab54s-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
          data-testid="ErrorOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-zioonp-MuiAlert-message"
      >
        error.contentLoad
      </div>
    </div>
  </li>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
  />
</DocumentFragment>
`;

exports[`Rendering test renders correctly while loading data 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
  >
    <div
      class="MuiListItemIcon-root css-1luiz03-MuiListItemIcon-root"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse sc-jRueeR bhwWFP css-1epbneo-MuiSkeleton-root"
      />
    </div>
    <div
      class="MuiListItemText-root sc-gjZUHa hWsfEH css-cfq8qh-MuiListItemText-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
        />
      </span>
    </div>
  </li>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
  />
</DocumentFragment>
`;
